### PR TITLE
Add Vendored Dependencies to Spec

### PR DESCRIPTION
**Pull request by khrm**
Modified rpm build script to autogenerate vendor deps in tekton.spec

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: tekton/rpmbuild/build.sh [hunk 1]
```diff
@@ -15,6 +15,19 @@ version=${version/v}
 
 sed "s/_VERSION_/${version}/" ${repospecfile} > ${TMPD}/tekton.spec
 
+sed -i '/bundled(golang/d' tekton.spec
```
Comment (khrm) [inline tekton/rpmbuild/build.sh]: Removing dependencies.

File: tekton/rpmbuild/build.sh [hunk 2]
```diff
@@ -15,6 +15,19 @@ version=${version/v}
...
+
+mapfile -t bundle < <(python -c "ver = None;
```
Comment (khrm) [inline tekton/rpmbuild/build.sh]: Getting dependencies from vendor/modules.txt

Similar to how origin-clients(oc) does. They use godep. 
Check here: https://apps.fedoraproject.org/packages/origin-clients/sources/spec/

File: tekton/rpmbuild/tekton.spec [hunk 3]
```diff
@@ -606,6 +662,9 @@ cp -a cli-%{version}/docs/man/man1/* %{buildroot}%{_mandir}/man1
 %{_mandir}/*
 
 %changelog
+* Tue Sep 26 2019 Khurram Baig <kbaig@redhat.com> 0.4.0
```
Comment (chmouel) [inline tekton/rpmbuild/tekton.spec]: weird i am getting a 
`warning: bogus date in %changelog: Tue Sep 26 2019 Khurram Baig <kbaig@redhat.com> 0.4.0`

when building the spec

but i am not sure what exeactly

Comment (khrm) [reply tekton/rpmbuild/tekton.spec]: Got it. Day is wrong. Haha.

Comment (chmouel) [reply tekton/rpmbuild/tekton.spec]: try using emacs's `rpm-add-changelog-entry` to add them and it should work (rpmlint should tell you the error too) https://github.com/stigbjorlykke/rpm-spec-mode/blob/master/rpm-spec-mode.el

Comment (khrm) [reply tekton/rpmbuild/tekton.spec]: Installed that package. Thanks. It works/ :)

**Pull request merged**